pkgs/initiator/initiator.go

```diff
@@ -0,0 +1,790 @@
+		}
+		dkgResults = append(dkgResults, *result)
+		if err := validatorPubKey.Deserialize(result.ValidatorPubKey); err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+		sharePubKey := &bls.PublicKey{}
+		if err := sharePubKey.Deserialize(result.SharePubKey); err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+		sharePks[result.OperatorID] = sharePubKey
+		depositShareSig := &bls.Sign{}
+		if err := depositShareSig.Deserialize(result.DepositPartialSignature); err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+		sigDepositShares[result.OperatorID] = depositShareSig
+		ownerNonceShareSig := &bls.Sign{}
+		if err := ownerNonceShareSig.Deserialize(result.OwnerNoncePartialSignature); err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+		ssvContractOwnerNonceSigShares[result.OperatorID] = ownerNonceShareSig
+		c.Logger.Debug("Received DKG result from operator", zap.Uint64("ID", result.OperatorID))
+	}
+	return dkgResults, &validatorPubKey, sharePks, sigDepositShares, ssvContractOwnerNonceSigShares, nil
+}
+
+func (c *Initiator) SendInitMsg(init *wire.Init, id [24]byte, operators []*wire.Operator) ([][]byte, error) {
+	sszInit, err := init.MarshalSSZ()
+	if err != nil {
+		return nil, err
+	}
+	initMessage := &wire.Transport{
+		Type:       wire.InitMessageType,
+		Identifier: id,
+		Data:       sszInit,
+	}
+	tsssz, err := initMessage.MarshalSSZ()
+	if err != nil {
+		return nil, err
+	}
+	sig, err := crypto.SignRSA(c.PrivateKey, tsssz)
+	if err != nil {
+		return nil, err
+	}
+	// Create signed init message
+	signedInitMsg := &wire.SignedTransport{
+		Message:   initMessage,
+		Signer:    0,
+		Signature: sig,
+	}
+	signedInitMsgBts, err := signedInitMsg.MarshalSSZ()
+	if err != nil {
+		return nil, err
+	}
+	results, err := c.SendToAll(consts.API_INIT_URL, signedInitMsgBts, operators)
+	if err != nil {
+		return nil, err
+	}
+	return results, nil
+}
+
+func (c *Initiator) SendExchangeMsgs(exchangeMsgs [][]byte, id [24]byte, operators []*wire.Operator) ([][]byte, error) {
+	mltpl, err := c.MakeMultiple(id, exchangeMsgs)
+	if err != nil {
+		return nil, err
+	}
+	mltplbyts, err := mltpl.MarshalSSZ()
+	if err != nil {
+		return nil, err
+	}
+	results, err := c.SendToAll(consts.API_DKG_URL, mltplbyts, operators)
+	if err != nil {
+		return nil, err
+	}
+	return results, nil
+}
+
+func (c *Initiator) SendKyberMsgs(kyberDeals [][]byte, id [24]byte, operators []*wire.Operator) ([][]byte, error) {
+	mltpl2, err := c.MakeMultiple(id, kyberDeals)
+	if err != nil {
+		return nil, err
+	}
+
+	mltpl2byts, err := mltpl2.MarshalSSZ()
+	if err != nil {
+		return nil, err
+	}
+	responseResult, err := c.SendToAll(consts.API_DKG_URL, mltpl2byts, operators)
+	if err != nil {
+		return nil, err
+	}
+	return responseResult, nil
+}
+
+func (c *Initiator) NewID() [24]byte { // random ID for each new DKG initiation
+	var id [24]byte
+	b := uuid.New()
+	copy(id[:12], b[:])
+	b = uuid.New()
+	copy(id[12:], b[:])
+	return id
+}
```

Comment:
I moved this function to crypto or util, it doesn't need anything from `Initiator` so doesn't need to be a part of it.

Reply:
yeah, I saw it at crypto, but got lost after merge, Ill move it to crypto again

Reply:
moved to crypto